lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/codeactions/AddFeature.java

```diff
@@ -57,14 +63,30 @@ public void doCodeAction(ICodeActionRequest request, List<CodeAction> codeAction
         Diagnostic diagnostic = request.getDiagnostic();
         DOMDocument document = request.getDocument();
         TextDocument textDocument = document.getTextDocument();
+
+        LibertyWorkspace ws = LibertyProjectsManager.getInstance().getWorkspaceFolder(document.getDocumentURI());    
+        if (ws == null) {
+            LOGGER.warning("Could not add quick fix for missing feature because could not find Liberty workspace for document: "+document.getDocumentURI());
+            return;
+        }
+
+        FeatureListNode flNode = ws.getFeatureListGraph().get(diagnostic.getSource());
+        if (flNode == null) {
```

Comment:
I can't help but think that the graph wouldn't return a null node in the quick fix if the diagnostic was made. The null safeguard seems proper though.

Reply:
I think it would probably only happen in the test path.